src/main/kotlin/com/atlassian/performance/tools/report/api/judge/ActionReport.kt

```diff
@@ -0,0 +1,16 @@
+package com.atlassian.performance.tools.report.api.judge
+
+import com.atlassian.performance.tools.jiraactions.api.ActionType
+import com.atlassian.performance.tools.report.api.junit.JUnitReport
+
+/**
+ * Action aware report
+ *
+ * @property action Action associated with this report.
+ * @property nonExceptional True if the failure in this report is expected sometimes as a part of a normal flow, false otherwise.
+ */
+interface ActionReport : JUnitReport {
```

Comment:
What's the point of the interface? How do we expect users to extend it?

Reply:
For flexibility and maintainability. It helps with TDD and mocks. It's a common practice especially with IoC

Reply:
We expect the users to extend it the same fashion as they would extend JUnitReport. We need to provide new value type, and we go for an interface because easier to test (mock for instance)

Reply:
Actually maintainability suffers, because it's automatically hard to add methods.
TDD and mocks did not improve, because `Verdict` already was flexible enough. It allowed adding actions trivially via the constructor.
Common practice doesn't mean "always promote all API to SPI".

Reply:
> We expect the users to extend it the same fashion as they would extend JUnitReport.

JUnit report had different implementations from the get-go. It displayed polymorphic behavior - 3 ways of completing a JUnit test.
What's the expected polymorphism here?